.circleci/config.yml

```diff
@@ -0,0 +1,63 @@
+# Java Maven CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-java/ for more details
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    working_directory: ~/repo
+    environment:
+      MAVEN_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - run: git submodule sync
+      - run: git submodule update --init
+
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "pom.xml" }}
+
+      - run:
+          name: Install dependencies
+          command: mvn dependency:go-offline
+
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+
+      - run:
+          name: Run tests
+          command: mvn integration-test
+          
+      - run:
+          name: Mocking release
+          command: .circleci/circlescript.sh
+
+  release:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    working_directory: ~/repo
+    environment:
+      MAVEN_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - run: git submodule sync
+      - run: git submodule update --init
+      - run:
+          name: Maven release
+          command: .circleci/circlescript.sh
+
+workflows:
+  version: 2
+  build-all:
+    jobs:
+      - build
+      - release:
+          requires:
+            - build
+          filters:
+            branches:
```

Comment:
Other agents release on a git tag. Think that would be simpler and less error prone to auto generating a version and tagging within CI/CD.

Reply:
See other comments about tags.